staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -820,6 +834,19 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 //  * If the server responds with a status: *errors.StatusError or *errors.UnexpectedObjectError
 //  * http.Client.Do errors are returned directly.
 func (r *Request) Do() Result {
+	if r.err != nil {
+		klog.V(4).Infof("Error in request: %v", r.err)
+		return Result{
+			err: &RequestConstructionError{r.err},
+		}
+	}
+
+	if err := r.requestPreflightCheck(); err != nil {
```

Comment:
I like the previous location for this better.  The only benefit I'm seeing here is gettnig ahead of the ratelimiter, but it opens us up to having to remember this construct if we ever have another Do or DoRaw.  You can make the error typed if you like, but this duplicated construct doesn't look like a net improvement..

Reply:
Restored the default approach. Coming up with the next commit.